azurerm/resource_arm_sql_failover_group.go

```diff
@@ -0,0 +1,357 @@
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, serverName, name)
+	if err != nil {
+		return fmt.Errorf("Error issuing get request for SQL Failover Group %q (Resource Group %q, Server %q): %+v", name, resourceGroup, serverName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmSqlFailoverGroupRead(d, meta)
+}
+
+func resourceArmSqlFailoverGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).sqlFailoverGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	name := id.Path["failoverGroups"]
+
+	resp, err := client.Get(ctx, resourceGroup, serverName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request for SQL Failover Group %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("server_name", serverName)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.FailoverGroupProperties; props != nil {
+		if err := d.Set("read_write_endpoint_failover_policy", flattenReadWritePolicy(props.ReadWriteEndpoint)); err != nil {
+			return fmt.Errorf("Error setting `read_write_endpoint_failover_policy`: %+v", err)
+		}
+
+		if err := d.Set("readonly_endpoint_failover_policy", flattenReadOnlyPolicy(props.ReadOnlyEndpoint)); err != nil {
+			return fmt.Errorf("Error setting `read_only_endpoint_failover_policy`: %+v", err)
+		}
+
+		d.Set("databases", set.FromStringSlice(*props.Databases))
+		d.Set("role", string(props.ReplicationRole))
+
+		if err := d.Set("partner_servers", flattenPartnerServers(props.PartnerServers)); err != nil {
+			return fmt.Errorf("Error setting `partnet_servers`: %+v", err)
+		}
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmSqlFailoverGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).sqlFailoverGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	name := id.Path["failoverGroups"]
+
+	future, err := client.Delete(ctx, resourceGroup, serverName, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting SQL Failover Group %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error deleting SQL Failover Group %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return err
+}
+
+func expandReadWritePolicy(d *schema.ResourceData) *sql.FailoverGroupReadWriteEndpoint {
```

Comment:
Could we rename this to:
```suggestion
func expandSqlFailoverGroupReadWritePolicy(d *schema.ResourceData) *sql.FailoverGroupReadWriteEndpoint {
```

as its package wide?